### ISSUE TITLE
"Failed to initialize libp2p rolling file" when `--logfile` is specified

### ISSUE THREAD

**Issue by jimmygchen** [sigp/lighthouse#6246]
## Description

A discord user reported getting `Failed to initialize libp2p rolling file appender` on startup on his beacon nodes for several versions now.

The user uses the `--logfile` flag, which is not only used as the lighthouse log _file_ path, but also used as a _directory_ path for libp2p logs.

This causes the libp2p `RollingFileAppender` to fail to build, because the file already exist and directory cannot be created.

The code below assumes that `logfile` is a directory, and is used as the directory path for libp2p logging:

https://github.com/sigp/lighthouse/blob/6856134ded14d8986c4ddd40abd3b26f87f09924/lighthouse/src/main.rs#L618-L630

This was a regression introduced in #4988 (v4.6.0) and impacts users running beacon node with the `--logfile` flag.

## Version

v5.3.0

## Present Behaviour

Error log `Failed to initialize libp2p rolling file appender` and no libp2p and discv5 created.

The only workaround now is to avoid using `--logfile` and have the logs stored under the data directory.

## Expected Behaviour

No error and libp2p & discv5 logs present in the specified log location.
How _should_ the application behave?

## Steps to resolve

Please describe the steps required to resolve this issue, if known.

**Comment by Scott-Sanderson** [comment]
I am the discord user who reported the libp2p log message. 

Confirmed on my server that removing the `--logfile` flag makes the libp2p message go away.

**Comment by ackintosh** [comment]
What I think the application should do is to output the libp2p and discv5 logs to the parent directory of the path specified by `--logfile`. For example, if `/home/ackintosh/logs/beacon.log` is secified as `--logfile`, the libp2p and discv5 logs should go into the `/home/ackintosh/logs` directory.

Additionally, a cleaner and more ideal solution might be to add a `--log-dir` CLI parameter and remove `--logfile`, although this would be a breaking change.

**Comment by eserilev** [comment]
Going to pick this one up

**Comment by chong-he** [comment]
Discord user @ hernanirvaz reported seeing `Failed to initialize libp2p rolling file appender` on the VC. 

Upon testing, I found that with only the VC is starting, the files `discv5.2024-08-26.log` and `libp2p.2024-08-26.log` will be generated under `$datadir/beacon/logs`. From @michaelsproul https://discord.com/channels/605577013327167508/605577013331361793/1277404042557128714, with only VC running, these files should not be generated.

Because of this, when the VC `--datadir` is having a directory that the user running Lighthouse (e.g., using systemd) has no permission, it will log `Failed to initialize libp2p rolling file appender`, because the VC will try to create the files `discv` and `libp2p`:

```
INFO Logging to file                         path: "/var/lib/lighthouse/validators/logs/validator.log"
Failed to initialize libp2p rolling file appender
INFO Lighthouse started                      version: Lighthouse/v5.3.0-d6ba8c3
INFO Configured for network                  name: mainnet
```

**Comment by chong-he** [comment]
This issue is resolved and closed by #6266
